independent-projects/arc/runtime/src/main/java/io/quarkus/arc/impl/ArcContainerImpl.java

```diff
@@ -694,7 +694,15 @@ private static Set<InjectableBean<?>> resolve(List<InjectableBean<?>> matching)
         nonDefault.removeIf(InjectableBean::isDefaultBean);
         if (nonDefault.isEmpty()) {
             // All the matching beans were default
-            return Set.copyOf(matching);
+            // Sort them by priority, uses 0 when no priority was defined
+            List<InjectableBean<?>> priorityDefaultBeans = new ArrayList<>(matching);
+            priorityDefaultBeans.sort(ArcContainerImpl::compareDefaultBeans);
+            Integer highest = priorityDefaultBeans.get(0).getPriority();
+            priorityDefaultBeans.removeIf(bean -> !highest.equals(bean.getPriority()));
```

Comment:
Given the fact that we still try to minimize the usage of lamdas in the runtime I'd propose to use iterator here and `removeIf` [here](https://github.com/quarkusio/quarkus/pull/37075/files#diff-059855113c2695306641fb4168e85fe886efbf257a54a6872ec4d489b4cc5e5bR529) ;-)

Reply:
@mkouba  I actually used this for consistency sake - the exact same lambda approach is few lines below for priority of alternatives :)

Reply:
I see. Up to you. Either keep it or change both lambdas ;-)

Reply:
Both are now changed